core/templates/pages/exploration-editor-page/editor-tab/services/responses.service.ts

```diff
@@ -120,6 +125,19 @@ angular.module('oppia').factory('ResponsesService', [
       }
     };
 
+    var _reduceRuleIndexByOne = function(answerGroupIndex, ruleIndex) {
+      _answerGroups[answerGroupIndex].rules[ruleIndex].inputs.x--;
+      _saveAnswerGroups(_answerGroups);
+      _multiChoiceInputAnswerGroupRequiresUpdation = true;
+    };
+
+    var _makeRuleInvalid = function(answerGroupIndex, ruleIndex) {
+      _answerGroups[answerGroupIndex].rules[ruleIndex].inputs.x =
+      Number.MAX_SAFE_INTEGER;
```

Comment:
Please format as:

```
      _answerGroups[answerGroupIndex].rules[ruleIndex].inputs.x = (
        Number.MAX_SAFE_INTEGER);
```

Also -- why Number.MAX_SAFE_INTEGER? Seems a bit random. What did it use to be, previously?

Also is it possible to save an exploration when a rule for an interaction is invalid?

Reply:
In line 105 in `multiple-choice-input-validation.service.ts`, For an answer group to be invalid.
The answer index should be greater than the number of choices. So In order to cover all cases, I used `Number.MAX_SAFE_INTEGER`.(this ensures that the deleted answer group  always remains invalid even if new choices are added)